core/src/main/java/org/apache/druid/timeline/VersionedIntervalTimeline.java

```diff
@@ -307,20 +315,18 @@ public boolean isEmpty()
     );
   }
 
-  public Set<TimelineObjectHolder<VersionType, ObjectType>> findOvershadowed()
+  public Set<TimelineObjectHolder<VersionType, ObjectType>> findFullyOvershadowed()
   {
+    lock.readLock().lock();
```

Comment:
I think all of these methods should probably be calling `readLock().lock()` outside of the `try/finally` like you have done here, would you consider making them all consistent?

Reply:
Moved them outside of `try` block.